browser/brave_wallet/solana_provider_browsertest.cc

```diff
@@ -0,0 +1,179 @@
+    ASSERT_TRUE(https_server()->Start());
+
+    keyring_service_ =
+        KeyringServiceFactory::GetServiceForContext(browser()->profile());
+  }
+
+  content::WebContents* web_contents() {
+    return browser()->tab_strip_model()->GetActiveWebContents();
+  }
+
+  net::EmbeddedTestServer* https_server() { return &https_server_; }
+
+  HostContentSettingsMap* host_content_settings_map() {
+    return HostContentSettingsMapFactory::GetForProfile(browser()->profile());
+  }
+
+  void RestoreWallet() {
+    const char mnemonic[] =
+        "scare piece awesome elite long drift control cabbage glass dash coral "
+        "angry";
+    base::RunLoop run_loop;
+    keyring_service_->RestoreWallet(
+        mnemonic, "brave123", false,
+        base::BindLambdaForTesting([&run_loop](bool success) {
+          ASSERT_TRUE(success);
+          run_loop.Quit();
+        }));
+    run_loop.Run();
+  }
+
+  void AddAccount() {
+    base::RunLoop run_loop;
+    keyring_service_->AddAccount(
+        "Account 1", mojom::CoinType::SOL,
+        base::BindLambdaForTesting([&run_loop](bool success) {
+          ASSERT_TRUE(success);
+          run_loop.Quit();
+        }));
+    run_loop.Run();
+  }
+
+  void SetSelectedAccount(const std::string& address) {
+    base::RunLoop run_loop;
+    keyring_service_->SetSelectedAccount(
+        address, mojom::CoinType::SOL,
+        base::BindLambdaForTesting([&run_loop](bool success) {
+          EXPECT_TRUE(success);
+          run_loop.Quit();
+        }));
+    run_loop.Run();
+  }
+
+  void UserGrantPermission(bool granted) {
+    if (granted)
+      permissions::BraveWalletPermissionContext::AcceptOrCancel(
+          std::vector<std::string>{first_account}, web_contents());
+    else
+      permissions::BraveWalletPermissionContext::Cancel(web_contents());
+    ASSERT_EQ(EvalJs(web_contents(), "getConnectedAccount()",
+                     content::EXECUTE_SCRIPT_USE_MANUAL_REPLY)
+                  .ExtractString(),
+              granted ? first_account : "");
+  }
+
+  void CallSolanaConnect() {
+    ASSERT_TRUE(ExecJs(web_contents(), "solanaConnect()"));
+    base::RunLoop().RunUntilIdle();
+    ASSERT_TRUE(
+        brave_wallet::BraveWalletTabHelper::FromWebContents(web_contents())
+            ->IsShowingBubble());
+  }
+
+  void CallSolanaDisConnect() {
+    ASSERT_TRUE(EvalJs(web_contents(), "solanaDisconnect()",
+                       content::EXECUTE_SCRIPT_USE_MANUAL_REPLY)
+                    .ExtractBool());
+  }
+
+  bool IsSolanaConnected() {
+    return EvalJs(web_contents(), "isSolanaConnected()",
+                  content::EXECUTE_SCRIPT_USE_MANUAL_REPLY)
+        .ExtractBool();
+  }
+
+ private:
+  base::test::ScopedFeatureList scoped_feature_list_;
+  net::test_server::EmbeddedTestServer https_server_;
+  raw_ptr<KeyringService> keyring_service_ = nullptr;
+};
+
+IN_PROC_BROWSER_TEST_F(SolanaProviderTest, ConnectedStatusAndPermission) {
```

Comment:
Could we also cover the behavior of multiple tabs(different frames) in this file?

Reply:
added in d5242b1743014b073130dbd1c1ca850aae4a0c90